SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestTearOff/SpatialTestTearOff.cpp

```diff
@@ -0,0 +1,186 @@
+		{
+			AReplicatedTearOffActor* SpawnedTearOffActor = GetWorld()->SpawnActor<AReplicatedTearOffActor>(FVector(50.0f, 50.0f, 80.0f), FRotator::ZeroRotator, FActorSpawnParameters());
+			RegisterAutoDestroyActor(SpawnedTearOffActor);
+
+			// Make sure the Actor was correctly spawned
+			if (IsValid(SpawnedTearOffActor))
+			{
+				FinishStep();
+			}
+		});
+
+	// Allow for potential replication to propagate before checking that the SpawnedTearOffActor was not received.
+	AddStepFromDefinition(WorkerWaitForTimeStepDefinition, FWorkerDefinition::AllWorkers);
+
+	// Clients check that they have not received the newly spawned Actor
+	AddStep(TEXT("SpatialTestTearOffUniversalSpawnedActorReferenceSetup"), FWorkerDefinition::AllClients, nullptr, [this]()
+		{
+			TArray<AActor*> ReplicatedTearOffActors;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedTearOffActor::StaticClass(), ReplicatedTearOffActors);
+
+			// Only the StartupTearOffActor is expected to exist in the world at this point.
+			if (ReplicatedTearOffActors.Num() == 1)
+			{
+				FinishStep();
+			}
+		});
+
+	// Spawn a replicated Actor that does not call TearOff on BeginPlay().
+	AddStep(TEXT("SpatialTestTearOffServerSpawnReplicatedActor"), FWorkerDefinition::Server(1), nullptr, [this]()
+		{
+			SpawnedReplicatedActorBase = GetWorld()->SpawnActor<AReplicatedTestActorBase>(SpwanedReplicatedActorBaseLocation, FRotator::ZeroRotator, FActorSpawnParameters());
+			RegisterAutoDestroyActor(SpawnedReplicatedActorBase);
+			FinishStep();
+		});
+
+	// Make sure the spawned actor is correctly replicated and set a reference to it.
+	AddStep(TEXT("SpatialTestTearOffClientsSpawnedActorReferenceSetup"), FWorkerDefinition::AllClients, nullptr, nullptr, [this](float DeltaTime)
+		{
+			TArray<AActor*> ReplicatedActors;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedTestActorBase::StaticClass(), ReplicatedActors);
+
+			// Since ReplicatedTearOffActor inherits from AReplicatedTestActorBase, 2 AReplicatedTestActorBase are expected at this point:
+			// The start-up ReplicatedTearOfActor, and the newly spawned AReplicatedTestActorBase object.
+			if (ReplicatedActors.Num() == 2)
+			{
+				// Ensure the reference to the correct Actor is set.
+				if (ReplicatedActors[0]->IsA(AReplicatedTearOffActor::StaticClass()))
+				{
+					SpawnedReplicatedActorBase = Cast<AReplicatedTestActorBase>(ReplicatedActors[1]);
+				}
+				else
+				{
+					SpawnedReplicatedActorBase = Cast<AReplicatedTestActorBase>(ReplicatedActors[0]);
+				}
+
+				if (IsValid(SpawnedReplicatedActorBase) && SpawnedReplicatedActorBase->GetActorLocation().Equals(SpwanedReplicatedActorBaseLocation, 1.0f))
+				{
+					FinishStep();
+				}
+			}
+		}, 5.0f);
+
+	// Tear off the spawned replicated Actor and move it to a new location.
+	AddStep(TEXT("SpatialTestTearOffServerSetTearOffAndMoveActor"), FWorkerDefinition::Server(1), nullptr, [this]()
```

Comment:
Can you add an extra step showing you were able to replicate position before calling tear off ?

Reply:
Will do!